src/com/xilinx/rapidwright/rwroute/PartialRouter.java

```diff
@@ -353,17 +354,19 @@ protected void determineRoutingTargets() {
             // (it's possible for another connection to use a bounce node, but now that node is
             // needed as a site pin)
             for (Connection connection : netWrapper.getConnections()) {
-                for (RouteNode rnode : Arrays.asList(connection.getSinkRnode(), connection.getAltSinkRnode())) {
+                Consumer<RouteNode> action = (rnode) -> {
                     if (rnode == null) {
-                        continue;
+                        return;
                     }
                     RouteNode prev = rnode.getPrev();
                     if (prev == null) {
-                        continue;
+                        return;
                     }
                     stashedPrev.put(rnode, prev);
                     rnode.clearPrev();
-                }
+                };
+                action.accept(connection.getSinkRnode());
+                connection.getAltSinkRnodes().forEach(action);
```

Comment:
Previously, we had up to two nodes -- a primary sink and a solitary, optional, alternate sink.

This PR extends it to be a primary sink and a list of alternate sinks. Rather than duplicating the body of the lambda for both primary and alternate sinks (nor create a new list containing all of them) I hoisted it into a single lambda to be executed for the single sink and the list of sinks.